src/main/cli/cli.c

```diff
@@ -5917,6 +5977,9 @@ const clicmd_t cmdTable[] = {
     CLI_COMMAND_DEF("serialpassthrough", "passthrough serial data to port", "<id> [baud] [mode] ['reset']", cliSerialPassthrough),
 #endif
 #endif
+#ifdef USE_ESC_SENSOR
+CLI_COMMAND_DEF("escsensorpassthrough", "passthrough serial data to esc sensor port", "[baud]", cliEscSensorPassthrough),
```

Comment:
I think it would be better to, instead of adding this as a command of its own, add this as a special case of the existing `serialpassthrough` command, seeing that except for disabling the motors and configuring the motor pins this is a copy of what `serialpassthrough` already does.

Also, note that this is in the wrong position, as the list has to be alphabetical.

Reply:
ok i will try to make it an option of the general serialpassthrough.
as i dont know the use cases, i didnt want to mess it up ;)

Reply:
how should i add the needed options to the existing cliSerialPassthrough command?

its actual
`CLI_COMMAND_DEF("serialpassthrough", "passthrough serial data to port", "<id> [baud] [mode] ['reset']", cliSerialPassthrough),`

i would need to add the port options (to enable halfduplex) and i would need to add a special thing to make it find the esc_sensor port and to push up the motor output pins...

like

`CLI_COMMAND_DEF("serialpassthrough", "passthrough serial data to port", "<id> [baud] [mode] [options] [special_usage] ['reset']", cliSerialPassthrough),`

or like?

Reply:
i couldnt keep my fingers still :D, so i have put my needed/wanted changes on the actual master.

i added a options setting for both ports so the command will look like:
`    CLI_COMMAND_DEF("serialpassthrough", "passthrough serial data from port 1 to VCP / port 2", "<id1> [<baud1>] [<mode1>] [<none|options1>] [none|reset] [<id2>] [<baud2>] [<mode2>] [<none|options2>] ", cliSerialPassthrough),`

also instead of <id1> you could write "esc_sensor" and instead of <id2> "cli" to have it find the esc_sensor and cli port istelf.

i will test the reset with DTR thing later. for now it workes for my propose with:
"serialpassthrough esc_sensor 2000000 rxtx bidir_pp none cli 115200 rxtx none" command

Reply:
Sounds mostly reasonable. Using `esc_sensor` as a shorthand for 'the UART that is used for esc_sensor, and shut down the ESC output' is a good approach. Separating the port 1 options can be done too and makes it a bit cleaner.
Not sure why we need an explicit `cli` option (since everything is dictated by VCP in this case anyway), or options for the second port (since the options really apply to both ports combined)?

Reply:
i thought SERIAL_PORT_USB_VCP might not always fit, as for example on the KISSFCV2F7 target, the port where the USB UART is connected is USART4. but i will check if it workes without :)

for passthroughs between two ports where none is the cli passthrough the options might be needed? i can remove it if you like.

BTW. Travis gets en error because of pwmDisableMotors(); i already added define check USE_PWM_OUTPUT. but dont seems to help. any idea what i do wrong?
(undefined reference to `pwmDisableMotors')

Reply:
ok i now removed the "cli" thing for port1 and replaced SERIAL_PORT_USB_VCP with cliPort->identifier
`    serialPassthroughPort_t ports[2] = { {SERIAL_PORT_NONE, 0, 0, SERIAL_NOT_INVERTED, NULL}, {cliPort->identifier, 0, 0, SERIAL_NOT_INVERTED, cliPort} };
`
like that it workes fine for me with "just" writing:
"serialpassthrough esc_sensor 2000000 rxtx bidir_pp reset"

i also removed the pwmDisableMotors() as it is not needed because i change the pins to output PP means they wont follow any timer